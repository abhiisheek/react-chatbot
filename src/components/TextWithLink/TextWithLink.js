import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { extractLinkFunc } from '../utils/ExtractLink';
import withHyperLink from '../hocs/WithHyperLink';
import PlainText from '../PlainText';

/** Link extraction function (assumes the link to be at the end of the given string) */
const extractLinkFromProps = ({ data }) => extractLinkFunc(data);

const TextWithLinkComp = withHyperLink(extractLinkFromProps)(PlainText);

//TODO: Doc generator support only Class or functional components

/** Plain Text component with clickable link/hyper link */
class TextWithLink extends PureComponent {
  render() {
    return <TextWithLinkComp data={this.props.data} />
  }
}

TextWithLink.displayName = TextWithLinkComp.displayName;

TextWithLink.propTypes = {
  /** Text along with the link to be displayed */
  data: PropTypes.string.isRequired
};

export default TextWithLink;