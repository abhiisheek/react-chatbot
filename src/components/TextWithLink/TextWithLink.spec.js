import React from 'react';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';
import jsdom from 'jsdom';

import TextWithLink from './TextWithLink';

// For mount
const doc = jsdom.jsdom('<!doctype html><html><body></body></html>');
global.document = doc;
global.window = doc.defaultView;

describe('TextWithLink', () => {
  it('Should have displayName set on the component', () => {
    expect(TextWithLink.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer
      .create(<TextWithLink data="Help doc http://help.docs" />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Should render the given link in a Anchor tag', () => {
    const wrapper = mount(<TextWithLink data="Help doc http://help.docs" />);
    expect(wrapper.find('a').length).toBe(1);
    expect(wrapper.find('a').text()).toBe('http://help.docs');
    expect(wrapper
        .childAt(0)
        .type()
        .displayName.startsWith('PlainText')).toBeTruthy();
  });

  it('Should render without Anchor tag when link is not passed', () => {
    const wrapper = mount(<TextWithLink data="Help doc" />);
    expect(wrapper.find('a').length).toBe(0);
    expect(wrapper.children().length).toBe(1);
    expect(wrapper
        .childAt(0)
        .type()
        .displayName.startsWith('PlainText')).toBeTruthy();
  });
});
