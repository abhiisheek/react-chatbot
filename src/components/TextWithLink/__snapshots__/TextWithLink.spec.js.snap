// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithLink Snapshot test 1`] = `
<div>
  <div
    className="textMsg wrapper"
  >
    Help doc
  </div>
  <a
    href="http://help.docs"
    target="_blank"
  >
    http://help.docs
  </a>
</div>
`;
